src/App.jsx

```diff
@@ -14,45 +17,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const sortByABC = 'abc';
+const sortByLength = 'length';
+
+const sorting = (goodsArr, sortBy, isReversed) => {
+  const preparedGoods = [...goodsArr];
+
+  if (sortBy) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortBy) {
+        case sortByABC:
+          return good1.localeCompare(good2);
+
+        case sortByLength:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+};
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortMethod, setSortMethod] = useState('');
+  const [isReversed, setOrder] = useState(false);
+
+  const visibleGoods = sorting(goodsFromServer, sortMethod, isReversed);
```

Comment:
make sense to use useMemo here, because it's essential to performance

Reply:
Haven't studied useMemo yet. As I understand, it's not required for this task. Anyway, thanks for advice!